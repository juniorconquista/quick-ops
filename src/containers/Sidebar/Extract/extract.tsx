import React, { memo } from 'react';
import { Box, H1, P } from 'quarks-ui';

import { ExtractProps } from './extract.types';

const Extract: React.FC<ExtractProps> = (props) => {
    return (
        <Box styling="column" alignItems="flex-start" width="100%">
            <H1
                styling="base"
                color="text.contrast"
                fontWeight="bold"
                fontSize="12px"
                m="10px 15px"
            >
                Extrato
            </H1>
            <Box
                styling="row"
                height="1px"
                width="100%"
                border="solid 1px #dfe2e6"
                mb="15px"
                opacity=".6"
            />
            <Box styling="row" width="100%" justifyContent="space-between">
                <Box styling="column" m="0px 15px" alignItems="flex-start">
                    <H1
                        styling="base"
                        color="text.contrast"
                        fontWeight="normal"
                        fontSize="12px"
                    >
                        Descrição
                    </H1>
                    <H1
                        styling="base"
                        color="text.contrast"
                        fontWeight="bold"
                        fontSize="14px"
                        pt="10px"
                    >
                        Despesas declaradas
                    </H1>
                    <P styling="base" fontSize="11px" color="#9ea4ac">
                        Despesas declaradas pelo trooper
                    </P>
                    <H1
                        styling="base"
                        color="text.contrast"
                        fontWeight="bold"
                        fontSize="14px"
                        pt="15px"
                    >
                        Despesas aprovadas
                    </H1>
                    <P styling="base" fontSize="11px" color="#9ea4ac">
                        Despesas aprovadas pelo financeiro
                    </P>
                    <H1
                        styling="base"
                        color="text.contrast"
                        fontWeight="bold"
                        fontSize="14px"
                        pt="15px"
                    >
                        Pagamento realizado
                    </H1>
                    <P styling="base" fontSize="11px" color="#9ea4ac">
                        Pagamento realizado pelo financeiro
                    </P>
                </Box>
                <Box styling="column" m="0px 15px" alignItems="flex-start">
                    <H1
                        styling="base"
                        color="text.contrast"
                        fontWeight="normal"
                        fontSize="12px"
                        m="0px 15px"
                    >
                        Valor
                    </H1>
                    <H1
                        styling="base"
                        color="text.contrast"
                        fontWeight="bold"
                        fontSize="14px"
                        pt="10px"
                    >
                        {`${props.currencySymbol} ${props.declared}`}
                    </H1>
                    <H1
                        styling="base"
                        color="text.contrast"
                        fontWeight="bold"
                        fontSize="14px"
                        pt="25px"
                    >
                        {`${props.currencySymbol} ${props.approved}`}
                    </H1>
                    <H1
                        styling="base"
                        fontWeight="bold"
                        fontSize="14px"
                        pt="25px"
                        color="#4acc82"
                    >
                        {`${props.currencySymbol} ${props.approved}`}
                    </H1>
                </Box>
            </Box>
        </Box>
    );
};

export default memo(Extract);
