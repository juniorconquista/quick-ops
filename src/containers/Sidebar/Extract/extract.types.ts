export interface ExtractProps {
    currencySymbol: string;
    declared: number;
    approved: number;
}
