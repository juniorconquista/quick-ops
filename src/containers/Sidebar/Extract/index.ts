import ExtractItem from './extract';

export default ExtractItem;
