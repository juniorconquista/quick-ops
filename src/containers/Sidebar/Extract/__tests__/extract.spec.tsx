import React from 'react';
import { cleanup, render } from '@testing-library/react';

import ThemeProvider from '../../../../utils/ThemeProviderTest';
import Extract from '../extract';
import 'jest-styled-components';

const props = {
    // isOpen: false
    currencySymbol: 'R$',
    declared: 10,
    approved: 20,
};

describe('Extract container', () => {
    afterEach(() => {
        jest.clearAllMocks();
        cleanup();
    });

    it('tests basic snapshot', () => {
        const container = render(
            <ThemeProvider>
                <Extract {...props} />
            </ThemeProvider>,
        );
        expect(container.asFragment()).toMatchSnapshot();
    });
});
