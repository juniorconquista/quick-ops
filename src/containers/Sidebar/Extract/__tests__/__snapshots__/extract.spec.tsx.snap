// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Extract container tests basic snapshot 1`] = `
<DocumentFragment>
  .c1 {
  color: #f8f9fa;
  font-size: 2rem;
  color: #FFFFFF;
  font-weight: bold;
  font-size: 12px;
  margin: 10px 15px;
}

.c5 {
  color: #f8f9fa;
  font-size: 2rem;
  color: #FFFFFF;
  font-weight: normal;
  font-size: 12px;
}

.c6 {
  color: #f8f9fa;
  font-size: 2rem;
  color: #FFFFFF;
  font-weight: bold;
  font-size: 14px;
  padding-top: 10px;
}

.c8 {
  color: #f8f9fa;
  font-size: 2rem;
  color: #FFFFFF;
  font-weight: bold;
  font-size: 14px;
  padding-top: 15px;
}

.c9 {
  color: #f8f9fa;
  font-size: 2rem;
  color: #FFFFFF;
  font-weight: normal;
  font-size: 12px;
  margin: 0px 15px;
}

.c10 {
  color: #f8f9fa;
  font-size: 2rem;
  color: #FFFFFF;
  font-weight: bold;
  font-size: 14px;
  padding-top: 25px;
}

.c11 {
  color: #f8f9fa;
  font-size: 2rem;
  color: #4acc82;
  font-weight: bold;
  font-size: 14px;
  padding-top: 25px;
}

.c7 {
  color: #f8f9fa;
  font-size: 1rem;
  color: #9ea4ac;
  font-size: 11px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  width: 100%;
}

.c0:focus {
  outline: none;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  opacity: .6;
  margin-bottom: 15px;
  height: 1px;
  width: 100%;
  border: solid 1px #dfe2e6;
}

.c2:focus {
  outline: none;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
}

.c3:focus {
  outline: none;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0px 15px;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c4:focus {
  outline: none;
}

<div
    class="c0"
    width="100%"
  >
    <h1
      class="c1"
      color="text.contrast"
      font-size="12px"
      font-weight="bold"
    >
      Extrato
    </h1>
    <div
      class="c2"
      height="1px"
      opacity=".6"
      width="100%"
    />
    <div
      class="c3"
      width="100%"
    >
      <div
        class="c4"
      >
        <h1
          class="c5"
          color="text.contrast"
          font-size="12px"
          font-weight="normal"
        >
          Descrição
        </h1>
        <h1
          class="c6"
          color="text.contrast"
          font-size="14px"
          font-weight="bold"
        >
          Despesas declaradas
        </h1>
        <p
          class="c7"
          color="#9ea4ac"
          font-size="11px"
        >
          Despesas declaradas pelo trooper
        </p>
        <h1
          class="c8"
          color="text.contrast"
          font-size="14px"
          font-weight="bold"
        >
          Despesas aprovadas
        </h1>
        <p
          class="c7"
          color="#9ea4ac"
          font-size="11px"
        >
          Despesas aprovadas pelo financeiro
        </p>
        <h1
          class="c8"
          color="text.contrast"
          font-size="14px"
          font-weight="bold"
        >
          Pagamento realizado
        </h1>
        <p
          class="c7"
          color="#9ea4ac"
          font-size="11px"
        >
          Pagamento realizado pelo financeiro
        </p>
      </div>
      <div
        class="c4"
      >
        <h1
          class="c9"
          color="text.contrast"
          font-size="12px"
          font-weight="normal"
        >
          Valor
        </h1>
        <h1
          class="c6"
          color="text.contrast"
          font-size="14px"
          font-weight="bold"
        >
          R$ 10
        </h1>
        <h1
          class="c10"
          color="text.contrast"
          font-size="14px"
          font-weight="bold"
        >
          R$ 20
        </h1>
        <h1
          class="c11"
          color="#4acc82"
          font-size="14px"
          font-weight="bold"
        >
          R$ 20
        </h1>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
